google/cloud/aiplatform/_featurestores/featurestore.py

```diff
@@ -0,0 +1,457 @@
+                filters are case-sensitive.
+
+                -  ``feature_id``: Supports = comparisons.
+                -  ``description``: Supports = comparisons. Multi-token
+                   filters should be enclosed in quotes.
+                -  ``entity_type_id``: Supports = comparisons.
+                -  ``value_type``: Supports = and != comparisons.
+                -  ``labels``: Supports key-value equality as well as
+                   key presence.
+                -  ``featurestore_id``: Supports = comparisons.
+
+                Examples:
+
+                -  ``description = "foo bar"`` --> Any Feature with
+                   description exactly equal to ``foo bar``
+                -  ``value_type = DOUBLE`` --> Features whose type is
+                   DOUBLE.
+                -  ``labels.active = yes AND labels.env = prod`` -->
+                   Features having both (active: yes) and (env: prod)
+                   labels.
+                -  ``labels.env: *`` --> Any Feature which has a label
+                   with ``env`` as the key.
+
+                This corresponds to the ``query`` field
+                on the ``request`` instance; if ``request`` is provided, this
+                should not be set.
+            page_size (int):
+                Optional. The maximum number of Features to return. The
+                service may return fewer than this value. If
+                unspecified, at most 100 Features will be
+                returned. The maximum value is 100; any value
+                greater than 100 will be coerced to 100.
+            project (str):
+                Optional. Project to list features in. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional. Location to list features in. If not set, location
+                set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to list features. Overrides
+                credentials set in aiplatform.init.
+
+        Returns:
+            List[Features] - A list of managed feature resource objects
+        """
+        resource = cls._empty_constructor(
+            project=project, location=location, credentials=credentials
+        )
+
+        # Fetch credentials once and re-use for all `_empty_constructor()` calls
+        creds = initializer.global_config.credentials
+
+        search_features_request = {
+            "location": initializer.global_config.common_location_path(
+                project=project, location=location
+            ),
+            "query": query,
+        }
+
+        if page_size:
+            search_features_request["page_size"] = page_size
+
+        feature_list = (
+            resource.api_client.search_features(request=search_features_request) or []
+        )
+
+        return [
+            _featurestores.Feature(
```

Comment:
Similar to `_list`, this should use`cls._construct_sdk_resource_from_gapic` to avoid the additional get call on construction.

Reply:
Using `cls._construct_sdk_resource_from_gapic` in `Featurestore.search_features()` will return `Featurestore` class. Move `Featurestore.search_features()` to `Feature.search()` and utilize the class methods to retrieve searched feature list.